hashintel/hash
# H-1623, H-1490: Fix `/notes` page incorrectly displaying relative dates + sidebar updates

## 🌟 What is the purpose of this PR?




This PR fixes the timestamp headings on the notes page. Previously, if there was more than one note from more than a month ago, all but one of those notes would incorrectly be put under a "Over a year ago" heading.

It also updates the sidebar to reflect "Notes" instead of "Quick Note", and the updated icon for the notes page.

## 🔗 Related links





- H-1623
- H-1490

## 🔍 What does this change?




See commits/description

## Pre-Merge Checklist 🚀

### 🚢 Has this modified a publishable library?




This PR:


### 📜 Does this require a change to the docs?




The changes in this PR:


### 🕸️ Does this require a change to the Turbo Graph?




The changes in this PR:


## ⚠️ Known issues



## 🐾 Next steps



## 🛡 What tests cover this?



Manual testing.

## ❓ How to test this?



Testing this locally is cumbersome, as it involves creating notes and then spoofing their created at time on the `/notes` page. The way I did this is by:

1. Create more than 3 notes on the quick notes page
2. Replace line 133 - 139 in `notes.page.ts` with the following snippet, essentially giving all notes a created at time of over a month ago, where the day is incremented by one for each.

```TS
    let day = 0;

    return Object.values(latestQuickNoteEditionMap).map((quickNoteEntity) => {
      day++;

      return {
        quickNoteEntity,
        createdAt: new Date(`2023-11-0${day}T15:38:53.502Z`),
      };
    });
```

3. Remove the diff made in this PR on line 201, and inspect that the incorrect timestamp heading is displayed on the `/notes` page
4. Bring the diff back, and inspect the fix

## 📹 Demo



Previously:

<img width="2044" alt="image" src="https://github.com/hashintel/hash/assets/42802102/af3f83a0-ecfb-4b4c-b6b8-967bba74c9f4">

Now:

<img width="2045" alt="image" src="https://github.com/hashintel/hash/assets/42802102/47cea7ef-c6fd-4fb2-8c2b-988275bedac1">
